src/shared/logger/activation.ts

```diff
@@ -198,3 +202,62 @@ async function registerLoggerCommands(context: vscode.ExtensionContext): Promise
         )
     )
 }
+
+/**
+ * Deletes the older logs when there are too many or are using too much space.
+ */
+async function cleanLogFiles(): Promise<void> {
+    let files = await fs.readdir(logPath)
+
+    if (files.length > 100) {
+        deleteOldLogFiles(files, 10)
+        files = await fs.readdir(logPath)
+    }
+
+    let dirSize = 0
+    const oversizedFiles = []
+    for (const log of files) {
+        try {
+            const logSize = (await fs.stat(path.join(logPath, log))).size
+            dirSize += logSize
+            if (logSize > LOG_MAX_BYTES) {
+                oversizedFiles.push(log)
+            }
+        } catch (e) {
+            getLogger().error('Failed to access file details for %0', path.join(logPath, log), e)
+        }
+    }
+    // remove any single files over 100MB
+    if (oversizedFiles.length) {
+        for (const file of oversizedFiles) {
+            await fs.unlink(path.join(logPath, file))
+        }
+        files = await fs.readdir(logPath)
+    }
+    if (dirSize > LOG_MAX_BYTES) {
+        deleteOldLogFiles(files, 2)
+    }
+}
+
+/**
+ * Deletes the oldest created files, leaving the desired quantity of latest files.
+ */
+async function deleteOldLogFiles(files: string[], keepLatest: number): Promise<void> {
+    files.sort()
+    // This removes the latest files, leaving only the files to be deleted
+    files.length = files.length >= keepLatest ? files.length - keepLatest : 0
+    if (files.length) {
+        for (const file of files) {
+            try {
+                if ((await fs.stat(path.join(logPath, file))).isFile()) {
+                    await fs.unlink(path.join(logPath, file))
+                    getLogger().info(
+                        `Log folder contains more than 100 logs or is over 100MB. Deleted the ${files.length} oldest files`
```

Comment:
this is in the for-loop. should be outside of the loop, only printed once per session. But it should probably be done in `cleanLogFiles()`, which has more knowledge of the semantics.